### ISSUE TITLE
Is the auth parameter of socket.io currently supported? Because I added a parameter auth, which caused the Disconnection event to return "parse error"

### ISSUE THREAD

**Issue by WilliamGrant** [mrniko/netty-socketio#950]
netty-socketio server version: 2.0.6

Server

```java
package com.corundumstudio.socketio.demo;

import com.corundumstudio.socketio.*;

public class ServerConnectTest {

    public static void main(String[] args) throws InterruptedException {

        Configuration config = new Configuration();
        config.setTransports(Transport.WEBSOCKET);
        config.setHostname("192.168.0.24");
        config.setPort(8096);
        config.setNeedClientAuth(false);
        config.setPingInterval(10);
        config.setContext("/socket");
        config.setAuthorizationListener(handshakeData -> {
            System.out.println(handshakeData);
            return AuthorizationResult.SUCCESSFUL_AUTHORIZATION;
        });

        SocketIOServer socketIOServer = new SocketIOServer(config);
        socketIOServer.addConnectListener(client -> {
            System.out.print("************ Client is connected ************");
     
            client.sendEvent("connected", "You have successfully connected...");
        });

  
        socketIOServer.addDisconnectListener(client -> {
            System.out.print("************ Client has disconnected ***************");
        });

        socketIOServer.addEventListener( "DATA",
                Object.class,
                (SocketIOClient client, Object data, AckRequest ackRequest) -> {
                    System.out.println(data);
                });


       
        socketIOServer.start();
        System.out.println("Socket.IO server started");
        Thread.sleep(Integer.MAX_VALUE);

        socketIOServer.stop();
    }
}

```

Socket.IO client version: 4.7.0

Client

```js
import {EventEmitter} from 'events'
import {Socket as SIOSocket, connect as socketIOConnect} from 'socket.io-client'
import { WebSocket as WSSocket} from 'ws'

export class SocketTest extends EventEmitter {

    protected readonly timestampParam?: string
    
    protected client: SIOSocket | WSSocket



    constructor() {
        super()
    }

    public connect(url:string,path:string,port:number): void {
        this.client = socketIOConnect(url, {
            path: path,
            forceNew: true,
            autoConnect: false,
            reconnection: false,
            transports: ['websocket', 'polling'],
            timeout: 20000,
            timestampRequests: true,
            timestampParam: this.timestampParam,
            auth:{
                configured: true,
                id: 'ac111223-1b50-3cb8-a01f-df7948811b8d',
                version:'1.0.1',
                name: 'test_electron_device',
                sn: '2cda9d57-4c9d-4488-8023-649adb56dbac',
                key:'f485386a-4dd1-42b6-8c44-6d5d0c1ac7d8',
                bc:  'MFEPMA_TP',
                model:'Electric_DEVICE_001',
                isPaused: false,
                control: true,
                automation: true
            }
        })
        const sioClient: SIOSocket = this.client
        sioClient
            .on('connect_error', (connectError) => {
                sioClient.sendBuffer = []
                if (!this.destroyed) setTimeout(() => sioClient.connect(), 1000)
                this.emit('connect_error', connectError)
            })
            .on('disconnect', (reason) => {
                if (!this.destroyed) setTimeout(() => sioClient.connect(), 1000)
                this.emit('disconnect', reason)
            })
            .on('connect', () => {
                sioClient.sendBuffer = []
                this.emit('connect')
            })
            .connect()
    }
}
```

Expected behavior
![20240108180558](https://github.com/mrniko/netty-socketio/assets/12205634/ea4cdfbd-758e-47de-8a6c-9bced421368a)

I finds that the disconnect event in client and returns a parse error, and loops to request a socket connection, resulting in a loop prompting disconnect event (parse error). When I remove the auth parameter, this problem will not occur. I want to know how to solve this problem and will there be any feature updates after this?

**Comment by unverbraucht** [comment]
I ran into the same limitation and fixed it in my PR.

**Comment by mrniko** [comment]
Fixed in 2.0.7
